### PR TITLE
Add buttons for close tab, add tab, shuffle layouts, transfer layout and tearout layout

### PR DESCRIPTION
**Pull request by connormccafferty**

### CODE REVIEW

File: app/components/types.ts [hunk 1]
```diff
@@ -16,3 +38,7 @@ export type LayoutState = {
   layoutName: string;
   layout: OpenFin.LayoutOptions;
 };
+
+export type LayoutIdentity = {
```
Comment (gallak-openfin) [inline app/components/types.ts]: This should just be OpenFin.LayoutIdentity right?

File: app/components/move-layout.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+export async function transferLayout(layoutIdentity: LayoutIdentity): Promise<void> {
+    const originLayout = fin.Platform.Layout.wrapSync(layoutIdentity);
+    // Do not allow inter-window transfer or transfer between applications
+    if (layoutIdentity.name === fin.me.identity.name || layoutIdentity.uuid !== fin.me.identity.uuid) {
+        return;
+    }
+    // Get origin Layout views
+    const layoutViews = await originLayout.getCurrentViews();
+    // Re-parent origin Layout views target
```
Comment (connormccafferty) [inline app/components/move-layout.ts]:
```suggestion
    // Re-parent origin Layout views to target window.
```

File: app/components/move-layout.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    await fin.Platform.getCurrentSync().createWindow({ layoutSnapshot, reason: 'tearout' });
+}
+
+/**
+ * Move an existing Layout from another window to the current window.
+ * @param {LayoutIdentity} layoutIdentity Layout identity of the Layout to transfer to the current window
+ */
+export async function transferLayout(layoutIdentity: LayoutIdentity): Promise<void> {
+    const originLayout = fin.Platform.Layout.wrapSync(layoutIdentity);
+    // Do not allow inter-window transfer or transfer between applications
```
Comment (connormccafferty) [inline app/components/move-layout.ts]:
```suggestion
    // Do not allow inter-window transfer or transfer between applications,
```

File: app/components/move-layout.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    if (layoutIdentity.name === fin.me.identity.name || layoutIdentity.uuid !== fin.me.identity.uuid) {
+        return;
+    }
+    // Get origin Layout views
```
Comment (connormccafferty) [inline app/components/move-layout.ts]:
```suggestion
    // Get origin Layout views.
```

File: app/components/types.ts (hunk 1 above)
Comment (connormccafferty) [reply app/components/types.ts]: Didn't exist in our version of @openfin/core

File: public/multi-layout-large.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "runtime": {
     "arguments": "--inspect",
-    "version": "34.118.78.53"
+    "version": "alpha-v34"
```
Comment (connormccafferty) [inline public/multi-layout-large.json:4]: Will peg this to a version once build with https://gitlab.com/openfin/core/-/merge_requests/4399 is done.

File: public/multi-layout.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "runtime": {
     "arguments": "--inspect",
-    "version": "34.118.78.53"
+    "version": "alpha-v34"
```
Comment (connormccafferty) [inline public/multi-layout.json:4]: Will peg this to a version once build with https://gitlab.com/openfin/core/-/merge_requests/4399 is done.

File: public/multi-layout-large.json (hunk 5 above)
Comment (gallak-openfin) [reply public/multi-layout-large.json:4]: we also need this one https://gitlab.com/openfin/core/-/merge_requests/4397/pipelines

File: app/components/LayoutContainer.tsx [hunk 7]
```diff
@@ -52,13 +72,93 @@ export const LayoutContainer = () => {
...
+    if (layoutNameToTransfer) {
+      const platformWindows = await fin.Application.getCurrentSync().getChildWindows();
+      const platformWindowOpts = await Promise.all(platformWindows.filter(x => x.identity.name !== fin.me.identity.name).map(x => x.getOptions()));
+      // Naively grab the first match, layoutName isn't unique across platform windows.
+      const layoutWindowOpts = platformWindowOpts.find(x => Object.keys(x.layoutSnapshot.layouts).includes(layoutNameToTransfer));
+      if (layoutWindowOpts) {
+        const layoutIdentity = { uuid: fin.me.identity.uuid, name: layoutWindowOpts.name, layoutName: layoutNameToTransfer };
+        const originLayout = fin.Platform.Layout.wrapSync(layoutIdentity);
+        const layout = await originLayout.getConfig();
+        await transferLayout(layoutIdentity);
```
Comment (pbaize) [inline app/components/LayoutContainer.tsx:122]: Should add a comment here.

Comment (gallak-openfin) [reply app/components/LayoutContainer.tsx:122]: fixed

**Pull request merged**